### PR TITLE
Bar chart e2e test

### PR DESCRIPTION
**Pull request by rafawendel**
## What type of PR is this?

- [x] Other

## Description
According to the checklist proposed on issue [#5208](https://github.com/getredash/redash/issues/5208), adding chart end-to-end tests with Cypress is top priority test-wise.

This is a boilerplate of test that covers making a query and using it to generate a bar chart. The test will verify all tabs of `VisualizationEditor` and then plot a chart and verify that it is plotted by Plotly.

### Todos:

- [x] Create boilerplate
- [x] Add prints
- [ ] Add update test
- [x] Create final snapshot

### CODE REVIEW

**Review by gabrieldutra — COMMENTED** [review]
Although it's good to check the chart exists in the Dialog, this doesn't get us as much value as the Percy snapshots (they track both rendering differences and the non-existence of the Charts). So for this I was thinking about having the `createChartThroughUI` able to create all sorts of charts and having this spec like this:

```js
function createChartThroughUI(visualizationName, extraStepsFunction = () => {}) {
  // basic Chart setup with shared steps
  extraStepsFunction(); // may need better naming, but the idea is to configure extra options and assertions here
  // assert chart is rendered in the Dialog preview
  // save it
  return visualizationId; // this will be useful to later build a dashboard
}

it("creates Bar charts", function() {
  const basicBarChart = createChartThroughUI("Basic Bar Chart");
  const barChartWithSwappedAxes = createChartThroughUI("Bar Chart with Swapped Axes", () => {
    // swap axes
    // any extra ui assertions, e.g.: check it correctly swaps the Labels in the editor
  });
  // any other Bar Chart variations we want to cover with testing

  createDashboardWithVisualizations([basicBarChart, barChartWithSwappedAxes]).then(dashboardUrl => {
    cy.visit(dashboardUrl);
    cy.percySnapshot("Bar Chart");
  });
})
```

For building a dashboard with multiple visualizations there is an example with the [Pivot table](https://percy.io/Redash/redash/builds/7387748/view/421638468/1280?mode=diff&browser=chrome&snapshot=421638468):

image

I'm not sure if the code is perfect to be reused, but I'd love to help with that as I think building dashboards with different configured visualizations are a way to cover many things at once 🙂.

LMK your opinion

File: client/cypress/integration/visualizations/chart_spec.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
+/* global cy */
+
+import { getWidgetTestId } from "../../support/dashboard";
+import { get } from "lodash";
```
Comment (gabrieldutra) [inline client/cypress/integration/visualizations/chart_spec.js:4]:
```suggestion
```

File: client/cypress/integration/visualizations/chart_spec.js [hunk 2]
```diff
@@ -99,19 +98,23 @@ function assertAxesAndAddLabels(xaxisLabel, yaxisLabel) {
...
+
+    const widgets = [];
+    chartGetters.forEach((chartGetter, i) => {
+      const position = { autoHeight: false, sizeY: 8, sizeX: 3, col: (i % 2) * 3 };
+      cy.get(chartGetter).then(chart => {
+        widgets.push(cy.addWidget(dashboard.id, chart.id, { position }));
+      });
+    });
+
+    widgetsAssertionFn(widgets, dashboardUrl);
```
Comment (rafawendel) [inline client/cypress/integration/visualizations/chart_spec.js:117]: For some reason, this is not running 🤪. @gabrieldutra can you give me a little insight?

**Comment by rafawendel** [comment]
Well I found out the trouble was that my loop would fill the array after it ran (because the push ran inside a .then). I read about this class of problems in Cypress and there doesn't seem to be a good solution. The `cy.all` Gabriel added some time ago shares the same limitations.

What I did at the end of the day was combining `as` and `get` statements to achieve the desired asynchronous behavior. It's quite cumbersome to use, but it is what it is.

**Pull request closed without merging**